src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -378,6 +359,28 @@ private ConnectionState CreateConnection(HttpContext context)
             }
         }
 
+        private async Task<IList<Message>> ExtractMessages(HttpContext context, byte[] buffer)
```

Comment:
I could inline it back. I only extracted the method because the code is more readable. I could also use ValueTask<T> if it makes it better.